applications/nrf_desktop/src/modules/ble_qos.c

```diff
@@ -0,0 +1,689 @@
+{
+	struct chmap_filter_params chmap_params;
+	struct config_fetch_event *fetch_event;
+	struct params_ble *qos_ble_params;
+
+	fetch_event = new_config_fetch_event(sizeof(struct params_ble));
+	__ASSERT_NO_MSG(fetch_event != NULL);
+	qos_ble_params = (struct params_ble *) fetch_event->dyndata.data;
+
+	chmap_filter_params_get(mp_chmap_inst, &chmap_params);
+
+	qos_ble_params->min_channel_count = chmap_params.min_channel_count;
+	qos_ble_params->weight_crc_ok = chmap_params.ble_weight_crc_ok;
+	qos_ble_params->weight_crc_error = chmap_params.ble_weight_crc_error;
+	qos_ble_params->ble_block_threshold = chmap_params.ble_block_threshold;
+	qos_ble_params->eval_max_count = chmap_params.eval_max_count;
+	qos_ble_params->eval_duration = chmap_params.eval_duration;
+	qos_ble_params->sample_count_min =
+		chmap_params.maintenance_sample_count;
+	qos_ble_params->eval_keepout_duration =
+		chmap_params.eval_keepout_duration;
+	qos_ble_params->eval_success_threshold =
+		chmap_params.eval_success_threshold;
+
+	return fetch_event;
+}
+
+static struct config_fetch_event *fetch_event_qos_wifi_params(void)
+{
+	struct chmap_filter_params chmap_params;
+	struct config_fetch_event *fetch_event;
+	struct params_wifi *qos_wifi_params;
+
+	fetch_event = new_config_fetch_event(sizeof(struct params_wifi));
+	__ASSERT_NO_MSG(fetch_event != NULL);
+	qos_wifi_params = (struct params_wifi *) fetch_event->dyndata.data;
+
+	chmap_filter_params_get(mp_chmap_inst, &chmap_params);
+
+	qos_wifi_params->wifi_rating_inc = chmap_params.wifi_rating_inc;
+	qos_wifi_params->wifi_active_threshold =
+		chmap_params.wifi_active_threshold;
+	qos_wifi_params->wifi_present_threshold =
+		chmap_params.wifi_present_threshold;
+
+	return fetch_event;
+}
+
+static struct config_fetch_event *fetch_event_qos_channel_map(void)
+{
+	struct config_fetch_event *fetch_event;
+	struct params_chmap *qos_chmap;
+
+	fetch_event = new_config_fetch_event(sizeof(struct params_chmap));
+	__ASSERT_NO_MSG(fetch_event != NULL);
+	qos_chmap = (struct params_chmap *) fetch_event->dyndata.data;
+
+	k_mutex_lock(&m_data_access_mutex, K_FOREVER);
+	memcpy(qos_chmap->chmap, m_current_chmap, sizeof(m_current_chmap));
+	k_mutex_unlock(&m_data_access_mutex);
+
+	return fetch_event;
+}
+
+static struct config_fetch_event *fetch_event_qos_blacklist(void)
+{
+	struct config_fetch_event *fetch_event;
+	struct params_blacklist *qos_blacklist;
+
+	fetch_event = new_config_fetch_event(sizeof(struct params_blacklist));
+	__ASSERT_NO_MSG(fetch_event != NULL);
+	qos_blacklist = (struct params_blacklist *) fetch_event->dyndata.data;
+	memset(qos_blacklist, 0, sizeof(struct params_blacklist));
+
+	qos_blacklist->wifi_chn_bitmask = chmap_filter_wifi_blacklist_get();
+
+	return fetch_event;
+}
+
+static void update_parameters(
+	struct params_ble *p_qos_ble_params,
+	struct params_wifi *p_qos_wifi_params)
+{
+	struct chmap_filter_params *p_params;
+
+	p_params = k_malloc(sizeof(struct chmap_filter_params));
```

Comment:
Yes, this works too. Will add a static param struct and static bool to indicate validity

Reply:
Using dynamic memory allocation could take longer time and it may require to increase heap size. Anyway I will not insist on using static struct here. @pdnuaj, what do you think about this?